core/src/main/java/com/google/errorprone/bugpatterns/SameNameButDifferent.java

```diff
@@ -96,20 +96,21 @@ private boolean shouldIgnore() {
 
       private void handle(Tree tree) {
         if (tree instanceof IdentifierTree
-            && ((IdentifierTree) tree).getName().contentEquals("Buiilder")) {
+            && ((IdentifierTree) tree).getName().contentEquals("Builder")) {
           return;
         }
         String treeSource = state.getSourceForNode(tree);
+        if (treeSource == null) {
+          return;
+        }
         Symbol symbol = getSymbol(tree);
         if (symbol instanceof ClassSymbol) {
           List<TreePath> treePaths = table.get(treeSource, symbol.type.tsym);
           if (treePaths == null) {
             treePaths = new ArrayList<>();
             table.put(treeSource, symbol.type.tsym, treePaths);
           }
-          if (state.getEndPosition(tree) != Position.NOPOS) {
-            treePaths.add(getCurrentPath());
-          }
+          treePaths.add(getCurrentPath());
```

Comment:
Dropped the `if`-statement because `treeSource != null` implies `state.getEndPosition(tree) != Position.NOPOS`.